src/include/storage/record_buffer.h

```diff
@@ -343,16 +343,24 @@ class RedoBuffer {
   /**
    * Flush all contents of the redo buffer to be logged out, effectively closing this redo buffer. No further entries
    * can be written to this redo buffer after the function returns.
-   * @param committed whether the transaction holding this RedoBuffer is committed
+   * @param flush_buffer whether the transaction holding this RedoBuffer should flush the its redo buffer
    */
-  void Finalize(bool committed);
+  void Finalize(bool flush_buffer);
 
   /**
    * @return a pointer to the beginning of the last record requested, or nullptr if no record exists.
    */
   byte *LastRecord() const { return last_record_; }
 
+  /**
+   * @return true if this buffer has previously flushed to the log manager
+   */
+  bool HasFlushed() const { return has_flushed_; }
+
  private:
+  // Flag to denote if this RedoBuffer has flushed records to the log manager already. Used to determine if we should
+  // write an abort record
```

Comment:
nit: Can you also write down the reason behind this? (i.e. if no records from the abort transaction is written out, there is no need to write the record out. In the case where there is, this helps the recovery pass to quickly discard changes, etc.)

Reply:
Done.